asserts/gpgkeypairmgr.go

```diff
@@ -23,15 +23,51 @@ import (
 	"bytes"
 	"errors"
 	"fmt"
+	"os"
 	"os/exec"
+	"path/filepath"
+	"strconv"
 	"strings"
+
+	"github.com/snapcore/snapd/osutil"
 )
 
+func ensureGPGHomeDirectory(homedir string) (string, error) {
+	real, err := osutil.RealUser()
+	if err != nil {
+		return "", err
+	}
+
+	uid, err := strconv.Atoi(real.Uid)
+	if err != nil {
+		return "", err
+	}
+
+	gid, err := strconv.Atoi(real.Gid)
+	if err != nil {
+		return "", err
+	}
+
+	if homedir == "" {
+		homedir = os.Getenv("SNAP_GNUPGHOME")
+	}
+	if homedir == "" {
+		homedir = filepath.Join(real.HomeDir, ".snap")
+	}
+
+	if err := osutil.MkdirAllChown(homedir, 0700, uid, gid); err != nil {
+		return "", err
+	}
+	return homedir, nil
+}
+
 func runGPGImpl(homedir string, input []byte, args ...string) ([]byte, error) {
-	general := []string{"-q"}
-	if homedir != "" {
-		general = append([]string{"--homedir", homedir}, general...)
+	homedir, err := ensureGPGHomeDirectory(homedir)
+	if err != nil {
+		return nil, err
 	}
+
+	general := []string{"--homedir", homedir, "-q", "--no-auto-check-trustdb"}
 	allArgs := append(general, args...)
 
 	cmd := exec.Command("gpg", allArgs...)
```

Comment:
This isn't a very strong preference, but I'm somewhat inclined to stick with gpg.  The OS direction is for gpg2 to become gpg once it's ready to be used as the default; this is the way it is in Debian unstable for instance, and possibly yakkety too though I haven't checked.  So for me, saying "gpg" is more like "system default GnuPG implementation" than "GnuPG 1".